### PR TITLE
[71] Validate settings.yaml

### LINKED ISSUE #71: Validate settings.yml
Idea: add a validation step of `pinanas` to:
- fail fast
- have better error messages

_Originally posted by @yscialom in https://github.com/yscialom/pinanas/issues/70#issuecomment-1334962164_

### PR DESCRIPTION
**Pull request by yscialom**
Fixes #71

### CODE REVIEW

File: src/utils/requirements.txt
[the remark this answers is not in the document]
Comment (yscialom) [reply src/utils/requirements.txt:1]: I agree, the whole structure of our project has become a tad confusing and deserves a refactoring. I usually prefer not to blend reworks and evolutions in the same PRs, that is why I created #108. This is my next priority.

File: src/utils/settings-validator/schema.json [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+            }
+          }
+        },
+        "applications": {
+          "type": "array",
+          "description": "Enabled applications",
+          "items": {
+            "type": "string",
+            "title": "Application",
+            "enum": [ "adguardhome", "heimdall", "jellyfin", "netdata", "nextcloud" ]
```
Comment (yscialom) [inline src/utils/settings-validator/schema.json:282]: duplicati

Comment (yscialom) [reply src/utils/settings-validator/schema.json]: Fixed in last commit

**Review by glevil — APPROVED** [review]

File: src/utils/settings-validator/schema.json [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+        },
+        "users": {
+          "type": "array",
+          "description": "Users",
+          "minItems": 1,
+          "items": {
+            "type": "object",
+            "title": "user",
+            "description": "User",
+            "required": [ "login", "password", "email" ],
```
[the remark this answers is not in the document]
Comment (yscialom) [reply src/utils/settings-validator/schema.json:53]: Non ça va il est pas obligatoire lui ;)

**Pull request merged**